homeassistant/components/glances/sensor.py

```diff
@@ -299,3 +315,39 @@ def native_value(self) -> StateType:
                 StateType, value[self._sensor_label][self.entity_description.key]
             )
         return cast(StateType, value[self.entity_description.key])
+
+
+class GlancesUptimeSensor(GlancesSensor):
+    """Implementation of a Glances Uptime sensor."""
+
+    _attr_native_value: datetime | None = None
+
+    def __init__(
+        self,
+        coordinator: GlancesDataUpdateCoordinator,
+        description: GlancesSensorEntityDescription,
+        sensor_label: str = "",
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, description, sensor_label)
+        self._attr_native_value = self.coordinator.get_uptime()
+
+    @property
+    def native_value(self) -> StateType | datetime:
+        """Return the state of the resources."""
+        return self._attr_native_value
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        uptime = self.coordinator.get_uptime()
+        if (
+            self._attr_native_value is None
+            or uptime is None
+            or
+            # Accept only changes of more than 10 x DEFAULT_SCAN_INTERVAL to avoid flapping
+            uptime - self._attr_native_value > DEFAULT_SCAN_INTERVAL * 10
```

Comment:
Why would this be needed? The coordinator only updates once a minute.

Reply:
The glances API provides a string representing a timespan (like  "uptime": "3 days, 10:25:20") but does not provide the absolute time it refers to. There is a sum of delays which varies each time the API is called (internal Glances poll loop, duration of the Api call, time inside Home assistant before converting back to an absolute time..) which means that the resulting datetime computed in the HA sensor varies from a few seconds to a few minutes depending on the configuration, for each call of the API.
It's not a big deal for an uptime sensor as it is typically measured in days or weeks rather than seconds or minutes.
But without this check to eliminate small changes, the sensor value changes every minute which makes the history UI unreadable and spams the database uselessly.

Reply:
I have moved the update frequency code to the `DataUpdateCoordinator`